src/caffe/test/test_gradient_based_solver.cpp

```diff
@@ -82,7 +96,7 @@ class GradientBasedSolverTest : public MultiDeviceTest<TypeParam> {
        "        value: 1.0 "
        "      } "
        "      data_filler { "
-       "        type: 'gaussian' "
+       "        type: '" << (run.const_data ? "constant" : "gaussian") << "' "
```

Comment:
The `targets` top needs to always be Gaussian or else the test is ineffective. The constant vs. gaussian condition should apply to the filler for `data`. Switching this reveals that the `CheckAccumulation` test is broken since it doesn't set constant data. More ominously, it reveals that all the `TestLeastSquares*` for multi-GPU. Let's follow up on #2870.